src/goto-symex/goto_symex.cpp

```diff
@@ -310,6 +314,26 @@ goto_symext::try_evaluate_constant_string(
   return try_get_string_data_array(s_pointer_opt->get(), ns);
 }
 
+optionalt<std::reference_wrapper<const constant_exprt>>
```

Comment:
`optionalt<reference_wrapper<T>>` -> `T*`

Reply:
It was suggested in a previous PR to use optional references as they provide some additional semantic information. I don't really mind either way but we should be consistent about it.

Reply:
I don't think there is a semantic difference, a reference wrapper is basically a pointer that can't be null, by making it optional you can get the empty optional which is basically equivalent to a null pointer, so it's just a verbose way of saying the same thing.

Reply:
Yes, I agree that they behave the same. What I meant was that when using an `optionalt` it's clear from the signature of the method that it returns an optional value, whereas when using a plain pointer that's not necessarily clear.

Reply:
I suppose I would always assume a pointer might be null, otherwise you'd use a reference

Reply:
Sure, though a nullpointer return value could suggest that some kind of error has occured, whereas `optionalt` would suggest that this is part of normal and expected behavior.